daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/preprocessing/ValueTranslator.scala

```diff
@@ -89,156 +96,175 @@ private[engine] final class ValueTranslator(compiledPackages: CompiledPackages)
 
     val cids = Set.newBuilder[Value.ContractId]
 
-    def go(ty: Type, value: Value[ContractId], nesting: Int = 0): SValue =
+    def go(ty0: Type, value: Value[ContractId], nesting: Int = 0): SValue =
       if (nesting > Value.MAXIMUM_NESTING) {
         fail(s"Provided value exceeds maximum nesting level of ${Value.MAXIMUM_NESTING}")
       } else {
-        val newNesting = nesting + 1
-        (ty, value) match {
-          // simple values
-          case (TUnit, ValueUnit) =>
-            SValue.SUnit
-          case (TBool, ValueBool(b)) =>
-            if (b) SValue.SValue.True else SValue.SValue.False
-          case (TInt64, ValueInt64(i)) =>
-            SValue.SInt64(i)
-          case (TTimestamp, ValueTimestamp(t)) =>
-            SValue.STimestamp(t)
-          case (TDate, ValueDate(t)) =>
-            SValue.SDate(t)
-          case (TText, ValueText(t)) =>
-            SValue.SText(t)
-          case (TNumeric(TNat(s)), ValueNumeric(d)) =>
-            Numeric.fromBigDecimal(s, d).fold(fail, SValue.SNumeric)
-          case (TParty, ValueParty(p)) =>
-            SValue.SParty(p)
-          case (TContractId(_), ValueContractId(c)) =>
-            cids += c
-            SValue.SContractId(c)
-
-          // optional
-          case (TOptional(elemType), ValueOptional(mb)) =>
-            SValue.SOptional(mb.map((value: Value[ContractId]) => go(elemType, value, newNesting)))
-
-          // list
-          case (TList(elemType), ValueList(ls)) =>
-            SValue.SList(ls.map((value: Value[ContractId]) => go(elemType, value, newNesting)))
-
-          // textMap
-          case (TTextMap(elemType), ValueTextMap(entries)) =>
-            SValue.SGenMap(
-              isTextMap = true,
-              entries = entries.iterator.map { case (k, v) =>
-                SValue.SText(k) -> go(elemType, v, newNesting)
-              },
-            )
-
-          // genMap
-          case (TGenMap(keyType, valueType), ValueGenMap(entries)) =>
-            SValue.SGenMap(
-              isTextMap = false,
-              entries = entries.iterator.map { case (k, v) =>
-                go(keyType, k, newNesting) -> go(valueType, v, newNesting)
-              },
-            )
+        def newNesting = nesting + 1
```

Comment:
why did you change this to a def instead of a val?

Reply:
reverted